node_package/tests/streamServerRenderedReactComponent.test.jsx

```
@@ -158,15 +191,133 @@ describe('streamServerRenderedReactComponent', () => {
-    expect(chunks[1].html).toMatch(/<script>[.\s\S]*Async Error[.\s\S]*<\/script>/);
+    expect(chunks[1].html).toMatch(/the server rendering errored:\\n\\nAsync Error/);
     expect(chunks[1].consoleReplayScript).toBe('');
     expect(chunks[1].hasErrors).toBe(true);
     expect(chunks[1].isShellReady).toBe(true);
   });
+
+  it.each(['asyncRenderFunction', 'renderFunction'])(
+    'streams a component from a %s that resolves to a React component',
+    async (componentType) => {
+      const { renderResult, chunks } = setupStreamTest({ componentType });
+      await new Promise((resolve) => renderResult.on('end', resolve));
```

ESLint

[error] Return values from promise executor functions cannot be read.

(no-promise-executor-return)